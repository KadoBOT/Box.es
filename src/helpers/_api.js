import R from 'ramda'

const getJson = R.curry((x, obj) => typeof obj[x] === 'function' ? obj[x]() : obj[x])

//helper that calls the api and retrieves the data
const fetchUrl = url => fetch(url).then(getJson('json'))
const item = R.prop('item')
const comment = R.prop('comments')
const fetchItem = R.composeP(item, fetchUrl)
const fetchComments = R.composeP(comment, fetchUrl)

export const getItem = () => fetchItem('https://api.box.es/api/users/47776/items/278878.json')
export const getComments = () => fetchComments('https://api.box.es/api/users/47776/items/278878/comments.json?page=1')
