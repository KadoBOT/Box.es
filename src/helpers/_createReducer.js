import R from 'ramda'

//function to make it easier to creat redux reducers
export const createReducer = (init, handlers) =>
  (state = init, action) =>
    R.propOr(R.identity, R.prop('type', action), handlers)(state, action)
