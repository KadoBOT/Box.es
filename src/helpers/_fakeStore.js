//creates a fake store to be used inside the tests
export const fakeStore = (state) => {
	return {
		default: () => {},
		subscribe: () => {},
		dispatch: () => {},
		getState: () => {
			return { ...state };
		},
	};
};
