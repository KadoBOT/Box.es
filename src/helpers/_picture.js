import R from 'ramda'

const largePic = R.prop('large')
const isMain = R.propEq('is_main', true)
const findLarge = R.propEq('large')
const findMain = R.find(isMain)

export const isMainPicture = R.compose(largePic, findMain) //find the is_main picture and retrieves the larger one
export const isSelected = R.equals //compare two keys
