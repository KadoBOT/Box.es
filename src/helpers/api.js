import R from 'ramda'

const getJson = R.curry((x, obj) => typeof obj[x] === 'function' ? obj[x]() : obj[x])

const fetchUrl = url => fetch(url).then(getJson('json'))
const item = R.prop('item')
const fetchItem = R.composeP(item, fetchUrl)
export const getItem = () => fetchItem('https://api.box.es/api/users/47776/items/278878.json')
