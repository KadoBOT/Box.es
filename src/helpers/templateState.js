export default {
  "isFetching": false,
  "name": "Diamond Select Marvel Select: Juggernaut Action Figure",
  "description": "A Diamond Select Release! Sculpted by Gentle Giant! The ranks of the best-selling Marvel Select line expand to include one of the most deadly mutants ever, the unstoppable Juggernaut. Sculpted by Gentle Giant and standing a massive 7\" tall, Juggernaut is the latest in the, in scale, over sized figures from the Marvel Select line. Charles Xavier's half brother is ready to join Magneto in the fight against the X-Men and anyone else that stands in their way. Featuring Marvel Selects standard 16 points of articulation, Juggernaut, and the sans-helmet chase variant figure, are the textbook definition of Unstoppable!",
  "price": 22.43,
  "location_details": "Genosha",
  "currency": {
    "symbol": "$",
  },
  "likes_count": 80,
  "user": {
    "nickname": "xavier",
    "name": "Charles Xavier",
    "avatar": {
      "small_url": "http://static.comicvine.com/uploads/original/7/71941/1489898-charles_xavier.jpg",
    },
  },
  "item_photos": [
    {
      "large": "https://images-na.ssl-images-amazon.com/images/I/719022GksTL._SL1500_.jpg",
      "small": "https://images-na.ssl-images-amazon.com/images/I/719022GksTL._SL1500_.jpg",
      "id": 473580,
      "is_main": false
    },
    {
      "large": "http://www.angelfire.com/mech/jinsaotome/Blog/Juggyreview_001.JPG",
      "small": "http://www.angelfire.com/mech/jinsaotome/Blog/Juggyreview_001.JPG",
      "id": 473579,
      "is_main": false
    },
    {
      "large": "https://actionfigurecanada.files.wordpress.com/2011/09/juggernaut-marvel-select-figure-2.jpg",
      "small": "https://actionfigurecanada.files.wordpress.com/2011/09/juggernaut-marvel-select-figure-2.jpg",
      "id": 473581,
      "is_main": true
    },
    {
      "large": "http://4.bp.blogspot.com/-qNdk8wZtZSY/UgGmCHgBLJI/AAAAAAAB0pk/2qpwZTuNSls/s1600/52458-1.jpg",
      "small": "http://4.bp.blogspot.com/-qNdk8wZtZSY/UgGmCHgBLJI/AAAAAAAB0pk/2qpwZTuNSls/s1600/52458-1.jpg",
      "id": 473582,
      "is_main": false
    },
    {
      "large": "https://s-media-cache-ak0.pinimg.com/originals/08/38/8f/08388f9c134afd90f567be95a4fe610b.jpg",
      "small": "https://s-media-cache-ak0.pinimg.com/originals/08/38/8f/08388f9c134afd90f567be95a4fe610b.jpg",
      "id": 473583,
      "is_main": false
    }
  ],
  "recent_comments": [
    {
      "id": 184543,
      "comment": "Miss those classic outfits ",
      "user": {
        "id": 13118,
        "nickname": "colossus",
        "name": "Piotr Rasputin",
        "avatar": {
          "small_url": "http://cdn2-www.comingsoon.net/assets/uploads/2015/12/deadpool-colossus-header.jpg",
          "is_default": false
        },
        "share_url": "http://box.es/nixon"
      }
    },
  ],
}
