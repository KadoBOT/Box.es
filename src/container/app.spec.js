import React from 'react'
import { Provider, connect } from 'react-redux'
import { shallow, render, mount } from 'enzyme';

import { fakeStore } from '../helpers/_fakeStore'
import ConnectedApp, { App } from './App'


it('should render the Live MainPage', () => {
  process.env.REACT_APP_ENV === 'notTemplate'

  const store = fakeStore({});
  const wrapper = shallow(
    <Provider store={store}>
      <ConnectedApp />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();

  process.env.REACT_APP_ENV === 'template'
});

it('should render the Template MainPage', () => {
  const store = fakeStore({});
  const wrapper = shallow(
    <Provider store={store}>
      <ConnectedApp />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
