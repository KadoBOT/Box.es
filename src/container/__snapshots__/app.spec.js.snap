exports[`test should render the Live MainPage 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Connect(MainPage) />,
  "nodes": Array [
    <Connect(MainPage) />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
                Object {
                        "default": [Function],
                        "dispatch": [Function],
                        "getState": [Function],
                        "subscribe": [Function],
                      }
        }>
        <Connect(MainPage) />
</Provider>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <Connect(MainPage) />,
          "store": Object {
            "default": [Function],
            "dispatch": [Function],
            "getState": [Function],
            "subscribe": [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "default": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "subscribe": [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Connect(MainPage) />,
        "_debugID": 2,
        "_renderedOutput": <Connect(MainPage) />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
        Object {
            "default": [Function],
            "dispatch": [Function],
            "getState": [Function],
            "subscribe": [Function],
          }
    }>
    <Connect(MainPage) />
</Provider>,
}
`;

exports[`test should render the Template MainPage 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Connect(MainPage) />,
  "nodes": Array [
    <Connect(MainPage) />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
                Object {
                        "default": [Function],
                        "dispatch": [Function],
                        "getState": [Function],
                        "subscribe": [Function],
                      }
        }>
        <Connect(MainPage) />
</Provider>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <Connect(MainPage) />,
          "store": Object {
            "default": [Function],
            "dispatch": [Function],
            "getState": [Function],
            "subscribe": [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "default": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "subscribe": [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Connect(MainPage) />,
        "_debugID": 4,
        "_renderedOutput": <Connect(MainPage) />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
        Object {
            "default": [Function],
            "dispatch": [Function],
            "getState": [Function],
            "subscribe": [Function],
          }
    }>
    <Connect(MainPage) />
</Provider>,
}
`;
